packages/store/src/internal/state-factory.ts

```diff
@@ -424,29 +408,20 @@ export class StateFactory implements OnDestroy {
   }
 
   private hydrateActionMetasMap({ path, actions, instance }: MappedStore): void {
-    const actionTypeToMetasMap = this.actionTypeToMetasMap;
-
     for (const actionType of Object.keys(actions)) {
-      // Initialize the map entry if it does not already exist for that
-      // action type. Note that action types may overlap between states,
-      // as the same action can be handled by different states.
-      if (!actionTypeToMetasMap.has(actionType)) {
-        actionTypeToMetasMap.set(actionType, []);
-      }
+      // This involves combining each individual action metadata with
+      // the state instance and the path—essentially everything needed
+      // to invoke an action. This eliminates the need to loop over states
+      // every time an action is dispatched.
+      const actionHandlers = actions[actionType].map(actionMeta => ({
+        ...actionMeta,
+        path,
+        instance
+      }));
```

Comment:
And this would change to create the handler that encapsulates the creation of the context and the invocation of the function on the instance:

```suggestion
      const actionHandlers = actions[actionType].map(actionMeta => (action: Instance<ActionType>) => {
        const stateContext = this._stateContextFactory.createStateContext(actionMeta.path);
        return actionMeta.instance[actionMeta.fn](stateContext, action);
      });
```
So then all that needs to be done on dispatch is for the handlers for that action type to be called with the action instance as an argument.
The try catch and result handling will be the same, but just around the simpler handler call.